pkg/reconciler/buildrun/resources/sources.go

```diff
@@ -25,19 +25,20 @@ const sourceTimestampName = "source-timestamp"
 // isLocalCopyBuildSource appends all "Sources" in a single slice, and if any entry is typed
 // "LocalCopy" it returns first LocalCopy typed BuildSource found, or nil.
 func isLocalCopyBuildSource(
-	build *buildv1alpha1.Build,
-	buildRun *buildv1alpha1.BuildRun,
-) *buildv1alpha1.BuildSource {
-	sources := []buildv1alpha1.BuildSource{}
-
-	sources = append(sources, build.Spec.Sources...)
-	sources = append(sources, buildRun.Spec.Sources...)
+	build *buildv1beta1.Build,
+	buildRun *buildv1beta1.BuildRun,
+) *buildv1beta1.Local {
+
+	// Note: In v1alpha1 we will append all sources across builds and buildruns, and then return
```

Comment:
Your code change does completely the same as the old code. Though, I wonder why we prefer the Build source over the BuildRun source. Should not the BuildRun source overwrite the Build source ?

Reply:
added a different approach.